print-babel.c

```diff
@@ -42,38 +42,45 @@
 #include "extract.h"
 
 
-static void babel_print_v2(netdissect_options *, const u_char *cp, u_int length);
+static void babel_print_v2(netdissect_options *, const u_char *cp, 
+                           u_int length, u_short bodylen);
 
 void
 babel_print(netdissect_options *ndo,
             const u_char *cp, u_int length)
 {
+    u_short bodylen;
     ndo->ndo_protocol = "babel";
     ND_PRINT("babel");
-
     ND_TCHECK_4(cp);
-
     if(GET_U_1(cp) != 42) {
         ND_PRINT(" invalid header");
         return;
     } else {
         ND_PRINT(" %u", GET_U_1(cp + 1));
     }
-
     switch(GET_U_1(cp + 1)) {
     case 2:
-        babel_print_v2(ndo, cp, length);
+        if (length < 4)
+            goto invalid;
+        bodylen = GET_BE_U_2(cp + 2);
+        ND_PRINT(" (%u)", bodylen);
```

Comment:
This prints the length twice; that should be done only by babel_print_v2().  That's causing some tests to fail.